packages/web/components/Modal/SettleDealModal/index.js

```diff
@@ -1,28 +1,48 @@
 import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { useForm } from 'react-hook-form';
+import { useRouter } from 'next/router';
 import { ReviewButton as Button } from '../CurateTechnologyModal/styles';
 import { toast } from '../../Toast';
 import { Modal, InfosContainer, Summary } from './styles';
 import { CurrencyInputField, InputField } from '../../Form';
 import { formatCurrencyToInt, formatMoney } from '../../../utils/helper';
+import { settleADeal } from '../../../services';
 
-const SettleDealModal = ({ closeModal }) => {
+const SettleDealModal = ({ closeModal, id }) => {
+	const router = useRouter();
 	const [totalValue, setTotalValue] = useState(0);
+	const [isSubmitting, setIsSubmitting] = useState(false);
 	const form = useForm();
 	const { watch } = form;
 	const formValues = watch();
 
-	const onSubmit = () => {
-		toast.success('Pedido fechado com sucesso');
+	const onSubmit = async () => {
+		setIsSubmitting(true);
+		const { quantity, unit_value } = formValues;
+		const { query } = router;
+
+		const result = await settleADeal(id, {
+			quantity,
+			unit_value: formatCurrencyToInt(unit_value),
+		});
+
+		if (result) {
+			toast.success('Pedido fechado com sucesso');
+		} else {
+			toast.error('Ocorreu um erro ao fechar pedido. Tente novamente mais tarde.');
```

Comment:
It would be interesting to display the error message that the server throws.

Reply:
@mateus4k it's a great idea for sure, but what about a refactoring in our API methods, so by default all requests returns the error and we don't need to return false in services?

This will be a massive refactoring, involving tests too. That's why I left this generic message. Any thoughts?